src/intents.cs

```diff
@@ -2245,25 +2232,21 @@ interface INRestaurantGuestDisplayPreferences : NSSecureCoding, NSCopying {
 
 	[Introduced (PlatformName.iOS, 10, 0)]
 	[Unavailable (PlatformName.MacOSX)]
-	[BaseType (typeof(NSObject))]
 	[DisableDefaultCtor]
-	interface INRestaurantMarketingPreferences : NSSecureCoding, NSCopying {
-
-		[Export ("initWithContactMethod:guestHasOptedInToMarketing:localizedDescriptionForMarketingEnabled:localizedDescriptionForMarketingNotEnabled:")]
-		[DesignatedInitializer]
-		IntPtr Constructor (INRestaurantMarketingPreferencesContactMethod contactMethod, bool guestHasOptedInToMarketing, string localizedDescriptionForMarketingEnabled, [NullAllowed] string localizedDescriptionForMarketingNotEnabled);
-
-		[Export ("contactMethod")]
-		INRestaurantMarketingPreferencesContactMethod ContactMethod { get; }
+	[BaseType (typeof (INIntentResolutionResult))]
+	interface INRestaurantGuestResolutionResult {
 
-		[Export ("guestHasOptedInToMarketing")]
-		bool GuestHasOptedInToMarketing { get; set; }
+		[Static]
+		[Export ("successWithResolvedRestaurantGuest:")]
+		INRestaurantGuestResolutionResult GetSuccess (INRestaurantGuest resolvedRestaurantGuest);
```

Comment:
If https://github.com/xamarin/xamarin-macios/pull/555/files#r73679252 turns out to be the way to go, we should apply the same thing for the 3 methods here.